### ISSUE TITLE
support circle

### ISSUE THREAD

**Issue by usefss** [lewish/asciiflow#229]
Ascii flow is awesome, but unfortunately, there are not many features, is there any way to support creating a circle??

**Comment by Qix-** [comment]
What would a circle look like?

**Comment by danielyrovas** [comment]
same as a square just rounding the corners

**Comment by usefss** [comment]
Now, I think that creating a circle in a grid is not that perfect, or what daniel said is not very practical. 
In the first place, The reason I think we need this is because when I am designing with a diagram tool I will use different shapes for different types of components, for example, squares for services, and circles for data storage and databases. In Asciiflow there is just one type of shape, if only there could be more somehow...

**Comment by danielyrovas** [comment]
yeah a diamond or squircle or circle would be pretty nice for UML or other flow diagrams

**Comment by porg** [comment]
## Additional function for all shape features
- All of these shapes (rect/square, oval/circle, triangle, diamon, etc) while being created/resized should work in conjunction with modifier keys to enforce/maintain a certain aspect ratio.

## Simply follow the conventions as usual in design apps.
- ALT down, then click, then drag → The first click is the center point.
- SHIFT down while creating object → Enforce 1:1 aspect ratio (circe, square, same-sided-triangle)
- SHIFT down while dragging corner of extsing object → Maintain given aspect ratio.

**Comment by porg** [comment]
Circle was a part of #71 which back then was turned down. But meanwhile the architecture may have come further.

**Comment by supertxtnet** [comment]
Svgbob has a nice specification for circles of various radii. https://ivanceras.github.io/content/Svgbob/Circles.html

It'd be great if you could use a circle tool to select the center and drag to increase the radius.

**Comment by porg** [comment]
I proposed [differentiated circle drawing modes](https://github.com/lewish/asciiflow/issues/229#issuecomment-1139597161) by using modifier keys. With them your desired workflow would work with:

> ALT down, then click, then drag → The first click is the center point.
